presentation/details/src/main/java/com/giraffe/details/screens/castDetails/CastDetailsRoute.kt

```
@@ -1,8 +1,32 @@
 package com.giraffe.details.screens.castDetails
 
 import androidx.navigation.NavController
-import com.giraffe.details.navigation.CastDetailsRoute
+import androidx.navigation.NavGraphBuilder
+import androidx.navigation.NavType
+import androidx.navigation.compose.composable
+import androidx.navigation.navArgument
+import com.giraffe.details.models.ReviewUI
 
-fun NavController.navigateToPersonDetails(personID: Int) {
-    navigate(CastDetailsRoute(personId = personID))
-}
\ No newline at end of file
+const val CAST_ROUTE = "castDetails"
+private const val CAST_ID_ARG = "personID"
+
+fun NavController.navigateToCastDetails(castID: Int) {
+    navigate("$CAST_ROUTE/$castID")
+}
+
+fun NavGraphBuilder.castDetailsRoute(
+    navController: NavController
+) {
+    composable(
+        route = "$CAST_ROUTE/{$CAST_ID_ARG}",
+        arguments = listOf(
+            navArgument(CAST_ID_ARG) {
+                type = NavType.IntType
+            })) { backStackEntry ->
+        val personId = backStackEntry.arguments?.getInt(CAST_ID_ARG) ?: 268
```

Don’t fall back to a hardcoded default for `personId`. If the argument is missing, the ViewModel should handle the error ,for example, by showing an error message or screen instead of `assuming` a specific ID.

---

done

---

We rehandle any things related to details route in other PR to make other PR it simple and small files